exchanges/huobi/huobi_websocket.go

```diff
@@ -75,427 +73,240 @@ var subscriptionNames = map[string]string{
+			return fmt.Errorf("error sending pong response: %w", err)
 		}
-
-		codes, _ := init.ErrorCode.(string)
-		return errors.New(h.Name + " Code:" + codes + " Message:" + init.ErrorMessage)
+		return nil
 	}
 
-	if init.ClientID > 0 {
-		if h.Websocket.Match.IncomingWithData(init.ClientID, respRaw) {
-			return nil
+	if action, err := jsonparser.GetString(respRaw, "action"); err == nil {
+		switch action {
+		case "ping":
+			return h.wsHandleV2ping(action, respRaw)
+		case wsSubOp, wsUnsubOp:
+			return h.wsHandleV2subResp(action, respRaw)
 		}
 	}
 
-	switch {
-	case strings.EqualFold(init.Op, authOp):
-		h.Websocket.SetCanUseAuthenticatedEndpoints(true)
-		// Auth captured
-		return nil
-	case strings.EqualFold(init.Topic, "accounts"):
-		var response WsAuthenticatedAccountsResponse
-		err := json.Unmarshal(respRaw, &response)
-		if err != nil {
-			return err
-		}
-		h.Websocket.DataHandler <- response
+	if err := getErrResp(respRaw); err != nil {
+		return err
+	}
 
-	case strings.Contains(init.Topic, "orders") &&
-		strings.Contains(init.Topic, "update"):
-		var response WsAuthenticatedOrdersUpdateResponse
-		err := json.Unmarshal(respRaw, &response)
-		if err != nil {
-			return err
-		}
-		data := strings.Split(response.Topic, ".")
-		if len(data) < 2 {
-			return errors.New(h.Name +
-				" - currency could not be extracted from response")
-		}
-		orderID := strconv.FormatInt(response.Data.OrderID, 10)
-		var oSide order.Side
-		oSide, err = stringToOrderSide(response.Data.OrderType)
-		if err != nil {
-			h.Websocket.DataHandler <- order.ClassificationError{
-				Exchange: h.Name,
-				OrderID:  orderID,
-				Err:      err,
-			}
-		}
-		var oType order.Type
-		oType, err = stringToOrderType(response.Data.OrderType)
-		if err != nil {
-			h.Websocket.DataHandler <- order.ClassificationError{
-				Exchange: h.Name,
-				OrderID:  orderID,
-				Err:      err,
-			}
-		}
-		var oStatus order.Status
-		oStatus, err = stringToOrderStatus(response.Data.OrderState)
-		if err != nil {
-			h.Websocket.DataHandler <- order.ClassificationError{
-				Exchange: h.Name,
-				OrderID:  orderID,
-				Err:      err,
-			}
-		}
-		var p currency.Pair
-		var a asset.Item
-		p, a, err = h.GetRequestFormattedPairAndAssetType(data[1])
-		if err != nil {
-			return err
-		}
-		h.Websocket.DataHandler <- &order.Detail{
-			Price:           response.Data.Price,
-			Amount:          response.Data.UnfilledAmount + response.Data.FilledAmount,
-			ExecutedAmount:  response.Data.FilledAmount,
-			RemainingAmount: response.Data.UnfilledAmount,
-			Exchange:        h.Name,
-			OrderID:         orderID,
-			Type:            oType,
-			Side:            oSide,
-			Status:          oStatus,
-			AssetType:       a,
-			LastUpdated:     time.Unix(response.TS*1000, 0),
-			Pair:            p,
+	if ch, err := jsonparser.GetString(respRaw, "ch"); err == nil {
+		s := h.Websocket.GetSubscription(ch)
+		if s == nil {
+			return subscription.ErrNotFound
```

Comment:
Can we please add the ch var to this return?

Reply:
Fixed https://github.com/gbjk/gocryptotrader/commit/32d533c15